src/TesApi.Web/TesEventMessage.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using TesApi.Web.Storage;
+
+namespace TesApi.Web
+{
+    /// <summary>
+    /// Represents the events sent by the node task runner.
+    /// </summary>
+    /// <remarks>This should be transient in DI.</remarks>
+    public class TesEventMessage
+    {
+        static TesEventMessage() => Tes.Utilities.NewtonsoftJsonSafeInit.SetDefaultSettings();
```

Comment:
I think we want to avoid new usages of NewtonSoft. The runner library has a method for deserialization. Maybe you can use it, make it public if needed. We should use only one method in the code base for deserialization and serialization of the class.